lib/puppet/type/cs_colocation.rb

```diff
@@ -73,20 +74,53 @@ def should=(value)
   end
 
   autorequire(:cs_shadow) do
-    [ @parameters[:cib] ]
+    autos = []
+    if @parameters[:cib]
+      autos << @parameters[:cib].value
+    end
+
+    autos
+  end
+
+  if Puppet::PUPPETVERSION.to_f >= 4.0
```

Comment:
what is the antipattern? the autonotify or the fact that I check the puppet version because it is not supported in Puppet 3?